core/runtime/jsruntime.rs

```diff
@@ -1221,6 +1225,42 @@ impl JsRuntime {
     Ok(js_runtime)
   }
 
+  /// If extensions were initialized with `lazy_init`, they need to be
+  /// fully initialized with this method.
+  pub fn lazy_init_extensions(
+    &self,
+    ext_args: Vec<ExtensionArguments>,
+  ) -> Result<(), CoreError> {
+    if ext_args.len() != self.inner.state.lazy_extensions.len() {
+      return Err(CoreError::ExtensionLazyInitCountMismatch(
+        ext_args.len(),
+        self.inner.state.lazy_extensions.len(),
+      ));
+    }
+
+    let mut state = self.inner.state.op_state.borrow_mut();
+
+    for (mut args, expected_name) in ext_args
+      .into_iter()
+      .zip(self.inner.state.lazy_extensions.iter())
+    {
+      if args.name != *expected_name {
+        return Err(CoreError::ExtensionLazyInitOrderMismatch(
+          expected_name,
+          args.name,
+        ));
+      }
+
+      let Some(f) = args.op_state_fn.take() else {
```

Comment:
When can it be `None`?

Reply:
if you have an extension with no `state` or `config`